prov/gni/src/gnix_ep.c

```diff
@@ -1371,10 +1412,46 @@ DIRECT_FN STATIC int gnix_ep_control(fid_t fid, int command, void *arg)
 					ret);
 				goto err;
 			}
-			if (ep->send_cq)
+
+			ret = _gnix_ep_enable(ep);
+			if (ret != FI_SUCCESS) {
+				GNIX_WARN(FI_LOG_EP_CTRL,
+				     "_gnix_ep_enable call returned %d\n",
+					ret);
+				goto err;
+			}
+
+			/*
+			 * set enable flags and add nic to poll lists
+			 */
```

Comment:
I don't understand the code that follows.  Isn't _gnix_ep_enable() doing exactly this?